system/Validation/FormatRules.php

```diff
@@ -346,4 +346,22 @@ public function valid_date(?string $str = null, ?string $format = null): bool
 
         return $date !== false && $errors !== false && $errors['warning_count'] === 0 && $errors['error_count'] === 0;
     }
+
+    /**
+     * Token JWT Validator
+     * 
+     * @param string $str String
+     */
+    public function valid_token(string $str = ''): bool
+    {
+        $newString = explode('.', $str);
+
+        if (count($newString) === 3) {
+            foreach ($newString as $key => $value) {
+                return base64_encode(base64_decode($value, true)) === $value;
+            }
```

Comment:
I think it should check all data to be valid, not only one that found valid early:

```suggestion
            foreach ($newString as $value) {
                $isValid = base64_encode(base64_decode($value, true)) === $value;
                if (! $isValid) {
                    return false;
                }
            }
            
            return true;
```

Reply:
also, `==` suffix seems need to be verified:

```bash
php > echo base64_encode(base64_decode('eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ', true));
eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ==
```

Reply:
should be something like this:

```suggestion
            foreach ($newString as $value) {
                $encoded = rtrim(base64_encode(base64_decode($value, true)), '==');
                $isValid =  $encoded === $value;
                if (! $isValid) {
                    return false;
                }
            }
            
            return true;
```